libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -107,7 +111,7 @@ func outbound*(p: PubSubPeer): bool =
   else:
     false
 
-proc recvObservers(p: PubSubPeer, msg: var RPCMsg) =
+proc recvObservers*(p: PubSubPeer, msg: var RPCMsg) =
```

Comment:
We could remove this from the pubsubpeer and directly handle it in pubsub
The reason it was here is just because the decoding happened to be here

Reply:
Should we also change it ti PubSub here?

```
  PubSubObserver* = ref object
    onRecv*: proc(peer: PubSubPeer; msgs: var RPCMsg) {.gcsafe, raises: [].}
```

Reply:
No, that would break the API (and wouldn't really make sense)

Reply:
This is called in the proc you mentioned.

Reply:
By removing this proc from `PubSubPeer` and directly handling it in PubSub, do you mean getting completely rid of the PubSubpeer instance? The `onRecv` call still requires `PubSubPeer` as an argument.

Reply:
I'm just talking about the observers, you can remove PubSubPeer.observers, and move this proc to PubSub which also holds the observers

Reply:
`observers` is also used in `send`. Should I move everything?

Reply:
Oh, I didn't realize
If not too bloody, yeah

Reply:
We'll work on that in a future PR.